### PR TITLE
Processor Mutex

### PR DESCRIPTION
**Pull request by passabilities**
It refactors the processor logic to be able to mutex baselet tables types (address, transaction and utxo)

### CODE REVIEW

**Comment by thehobbit85** [comment]
I changed this PR to 'Draft' until the blocking PR's are merged and it gets rebased.

File: src/common/utxobased/engine/makeUtxoEngineState.ts [hunk 1]
```diff
@@ -66,6 +66,7 @@ export function makeUtxoEngineState(
       processor
     })
     const percent = processedCount / totalCount
+    config.io.console.info('processed', percent)
```
Comment (passabilities) [inline src/common/utxobased/engine/makeUtxoEngineState.ts:69]: make sure is removed

File: src/common/utxobased/engine/makeUtxoEngineState.ts [hunk 2]
```diff
@@ -573,92 +561,111 @@ const processAddressUtxos = async (
...
+      newBalance = bs.add(newBalance, utxo.value)
     }
+    for (const id of currentUtxoIds) {
+      const utxo = await processor.removeUtxo(id)
+      newBalance = bs.sub(newBalance, utxo.value)
+    }
+
+    const diff = bs.sub(newBalance, oldBalance)
+    if (diff !== '0') {
+      console.log({ address, diff })
```
Comment (passabilities) [inline src/common/utxobased/engine/makeUtxoEngineState.ts:615]: make sure is removed

**Review by passabilities — COMMENTED** [review]
remove mutex from fetching

**Pull request merged**